connectors/spark/src/main/scala/io/unitycatalog/spark/ApiClientFactory.java

```diff
@@ -21,4 +23,38 @@ public static ApiClient createApiClient(URI url, String token) {
 
     return apiClient;
   }
+
+  public static ApiClient createApiClient(Configuration conf, URI url, String token) {
+    return createApiClient(conf, url, token, null);
+  }
+
+  public static ApiClient createApiClient(
+      Configuration conf,
+      URI url,
+      String token,
+      HttpRetryHandler retryHandler) {
```

Comment:
What's the case that we enabled the `retry` but don't have the `retryHandler` ?   I will think that people only need be aware either `retryEnabled` or `retryHandle`, but they don't need to perceive both, right ? 

Then for the above 
`
public static ApiClient createApiClient(Configuration conf, URI url, String token) {
    return createApiClient(conf, url, token, null);
  }
`

we only need to keep one public method, and remove the other one.  Does this make sense ?